charts/scalardl-auditor-monitoring/templates/_alloy-content.tpl

```diff
@@ -0,0 +1,99 @@
+{{/*
+Grafana Alloy components configuration
+*/}}
+{{- define "scalardl-auditor-monitoring.alloyContent" -}}
+discovery.kubernetes "scalardl_auditor" {
+  role = "pod"
+  namespaces {
+    names = ["{{ .Release.Namespace }}"]
+  }
+  selectors {
+    role = "pod"
+    label = "app.kubernetes.io/app=auditor"
+  }
+}
+
+{{- if .Values.global.envoy.enabled }}
+discovery.kubernetes "envoy" {
+  role = "pod"
+  namespaces {
+    names = ["{{ .Release.Namespace }}"]
+  }
+  selectors {
+    role = "pod"
+    label = "app.kubernetes.io/app=envoy"
+  }
+}
+{{- end }}
```

Comment:
This discovery configuration for Envoy is applied only when we set `global.envoy.enabled=true`.